__tests__/app.test.js

```diff
@@ -91,3 +91,64 @@ describe('GET /api/articles', () => {
     });
 })
 
+describe('POST /api/articles/:article_id/comments', () => {
+    test('resolves with status 201 and correct comment object.', () => {
+        const newComment = {
+            username: 'lurker',
+            body: 'not a bad read'
+        }
+        return request(app).post('/api/articles/1/comments').send(newComment).expect(201).then((res) => {
+            const comment = res.body.comment;
+            expect(comment).toEqual(newComment.body);
+        })
+    });
+
+    test('inserted comment is in database with correct keys and values', () => {
+        const newComment = {
+            username: 'lurker',
+            body: 'not a bad read'
+        }
+        return request(app).post('/api/articles/2/comments').send(newComment).then((res) => {
+            return db.query(
+                `SELECT * FROM comments
+                WHERE article_id = 2;`
+            ).then((res) => {
+                const comment = res.rows[0];
+                expect(comment.votes).toBe(0);
+                expect(comment).toEqual(expect.objectContaining({
+                  comment_id: expect.any(Number),
+                  body: expect.any(String),
+                  article_id: expect.any(Number),
+                  author: expect.any(String),
+                  votes: 0,
+                  created_at: expect.any(Date)
+                })
+                )
+            }) 
+
+        })
+    });
+
+    test('rejects with status 400 when not given required fields', () => {
+        const newComment = {
+            body: 'not a bad read'
+        }
+        return request(app).post('/api/articles/2/comments').send(newComment).expect(400);
```

Comment:
I would really prefer a message to come back along with the status code. We might be sending loads of 400s and 404s to the user for a variety of reasons. They benefit from knowing what that reason is. After all, they need to know they sent a bad body in order to correct it.